dbms/src/Common/ProcessCollector.h

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <ProcessMetrics/ProcessMetrics.h>
+#include <prometheus/counter.h>
+#include <prometheus/metric_family.h>
+#include <prometheus/registry.h>
+
+#include <mutex>
+
+namespace DB
+{
+
+class ProcessCollector : public prometheus::Collectable
```

Comment:
how about using `AsynchronousMetrics`?
just like https://github.com/pingcap/tiflash/blob/e0c82c01ca63d68f63cf8df7667f0e3b2429a43b/dbms/src/Interpreters/AsynchronousMetrics.cpp#L256-L263

Reply:
I'm worried that the original implementation may have an impact on `MetricsPrometheus::run`.

Reply:
And the collection frequency can be controlled by `AsynchronousMetrics::run()`.

Reply:
Async Metrics get metric every `95s`, it's too large(we want 5s)

Reply:
Looks `AsynchronousMetrics::run()` also controls the interval of push of gateway?
Also we want collect cpu metrics when grafana want to collect intead of a fixed interval

Reply:
ok